src/lib/hooks/swap/useSwapDefaults.ts

```diff
@@ -37,7 +37,7 @@ export default function useSwapDefaults({
   defaultOutputAddress,
   defaultOutputAmount,
 }: UseSwapDefaultsArgs) {
-  const updateSwap = useUpdateAtom(swapAtom)
+  const [swap, updateSwap] = useAtom(swapAtom)
```

Comment:
This adds a duplicate definition for `useAtomValue(swapAtom)`, so I don't think `swap` should be returned (because multiple codepaths doing the same thing leaves more room for error).

If you're worried about clarity, instead rename `useSwapDefaults` to `useSetSwapDefaults` so it is clear it only returns a setter.

Reply:
`useSet` is just so awful though

Reply:
`useSync` maybe